### PR TITLE
Worker get elastic run config from master

### PR DESCRIPTION
**Pull request by samplise**
### What changes were proposed in this pull request?

Before starting training, workers read elastic run configures from the master.

### Why are the changes needed?

We can distribute run configures from the master.

### Does this PR introduce any user-facing change?

No.

### How was this patch tested?

Unit test.

### CODE REVIEW

**Comment by workingloong** [comment]
Does the PR support updating the training configurations without restarting the job?

**Comment by samplise** [comment]
> Does the PR support updating the training configurations without restarting the job?

No. This pr allows to configure all jobs in a cluster with the same configuration in a simple way. Otherwise, we have to ask all users to have the same configuration in their dlrover-run command line.

**Review by majieyue — APPROVED** [review]
lgtm

File: dlrover/trainer/torch/elastic_run.py [hunk 1]
```diff
@@ -304,26 +304,46 @@ def _elastic_config_from_args(
...
+
+    master_config = _elastic_config_from_master(config)
     elastic_config = ElasticLaunchConfig(**config.__dict__)
 
     # PyTorch >= 2.3.0 remove log_dir in the LaunchConfig.
     if not version_less_than_230():
         elastic_config.log_dir = config.logs_specs.root_log_dir
+
     elastic_config.network_check = getattr(args, "network_check", False)
+    if master_config.network_check:
```
Comment (BalaBalaYi) [inline dlrover/trainer/torch/elastic_run.py:316]: No need to set 'false' if config from master is false?

Comment (samplise) [reply dlrover/trainer/torch/elastic_run.py:316]: Yes. Here we assume the local command has the highest priority.

**Review by BalaBalaYi — APPROVED** [review]
LGTM

**Pull request merged**